src/Categories/Functor/Displayed.agda

```diff
@@ -0,0 +1,68 @@
+{-# OPTIONS --safe --without-K #-}
+
+module Categories.Functor.Displayed where
+
+open import Function.Base using () renaming (id to id→; _∘_ to _∙_)
+open import Level
+
+open import Categories.Category
+open import Categories.Category.Displayed
+open import Categories.Functor
+
+private
+  variable
+    o ℓ e o′ ℓ′ e′ o″ ℓ″ e″ o‴ ℓ‴ e‴  : Level
+    B₁ B₂ B₃ : Category o ℓ e
+
+record DisplayedFunctor {B₁ : Category o ℓ e} {B₂ : Category o′ ℓ′ e′} (C : Displayed B₁ o″ ℓ″ e″) (D : Displayed B₂ o‴ ℓ‴ e‴) (F : Functor B₁ B₂) : Set (o ⊔ ℓ ⊔ e ⊔ o″ ⊔ ℓ″ ⊔ e″ ⊔ o‴ ⊔ ℓ‴ ⊔ e‴) where
+  private
+    module C = Displayed C
+    module D = Displayed D
+    module F = Functor F
+  field
+    F₀′ : ∀ {X} → C.Obj[ X ] → D.Obj[ F.₀ X ]
+    F₁′ : ∀ {X Y} {f : B₁ [ X , Y ]} {X′ Y′} → X′ C.⇒[ f ] Y′ → F₀′ X′ D.⇒[ F.₁ f ] F₀′ Y′
+    identity′ : ∀ {X} {X′ : C.Obj[ X ]} → F₁′ (C.id′ {X = X′}) D.≈[ F.identity ] D.id′
+    homomorphism′ : ∀ {X Y Z} {X′ : C.Obj[ X ]} {Y′ : C.Obj[ Y ]} {Z′ : C.Obj[ Z ]}
+                      {f : B₁ [ X , Y ]} {g : B₁ [ Y , Z ]} {f′ : X′ C.⇒[ f ] Y′} {g′ : Y′ C.⇒[ g ] Z′}
+                    → F₁′ (g′ C.∘′ f′) D.≈[ F.homomorphism ] F₁′ g′ D.∘′ F₁′ f′
+    F′-resp-≈[] : ∀ {X Y} {X′ : C.Obj[ X ]} {Y′ : C.Obj[ Y ]}
+                    {f g : B₁ [ X , Y ]} {f′ : X′ C.⇒[ f ] Y′} {g′ : X′ C.⇒[ g ] Y′}
+                  → {p : B₁ [ f ≈ g ]} → f′ C.≈[ p ] g′
+                  → F₁′ f′ D.≈[ F.F-resp-≈ p ] F₁′ g′
+
+  ₀′ = F₀′
+  ₁′ = F₁′
+
+id′ : ∀ {B : Category o ℓ e} {C : Displayed B o′ ℓ′ e′} → DisplayedFunctor C C id
+id′ {C = C} = record
+  { F₀′ = id→
+  ; F₁′ = id→
+  ; identity′ = Displayed.Equiv′.refl′ C
+  ; homomorphism′ = Displayed.Equiv′.refl′ C
+  ; F′-resp-≈[] = id→
+  }
+
+_∘F′_ : ∀ {C : Displayed B₁ o ℓ e} {D : Displayed B₂ o′ ℓ′ e′} {E : Displayed B₃ o″ ℓ″ e″}
+          {F : Functor B₂ B₃} {G : Functor B₁ B₂}
+        → DisplayedFunctor D E F → DisplayedFunctor C D G → DisplayedFunctor C E (F ∘F G)
```

Comment:
If you include the ones hidden by variables, this has 18 `Level` parameters! Is the most in the library (so far)?

Reply:
That is definitely, by a long shot, more than I've seen in any library.